src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Drivers/MicrosoftAccountSettingsDisplayDriver.cs

```diff
@@ -0,0 +1,84 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Http;
+using OrchardCore.DisplayManagement.Entities;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.DisplayManagement.Views;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Microsoft.Authentication.Settings;
+using OrchardCore.Microsoft.Authentication.ViewModels;
+using OrchardCore.Settings;
+
+namespace OrchardCore.Microsoft.Authentication.Drivers
+{
+    public class MicrosoftAccountSettingsDisplayDriver : SectionDisplayDriver<ISite, MicrosoftAccountSettings>
+    {
+        private readonly IAuthorizationService _authorizationService;
+        private readonly IDataProtectionProvider _dataProtectionProvider;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IShellHost _shellHost;
+        private readonly ShellSettings _shellSettings;
+
+        public MicrosoftAccountSettingsDisplayDriver(
+            IAuthorizationService authorizationService,
+            IDataProtectionProvider dataProtectionProvider,
+            IHttpContextAccessor httpContextAccessor,
+            IShellHost shellHost,
+            ShellSettings shellSettings)
+        {
+            _authorizationService = authorizationService;
+            _dataProtectionProvider = dataProtectionProvider;
+            _httpContextAccessor = httpContextAccessor;
+            _shellHost = shellHost;
+            _shellSettings = shellSettings;
+        }
+
+        public override async Task<IDisplayResult> EditAsync(MicrosoftAccountSettings settings, BuildEditorContext context)
+        {
+            var user = _httpContextAccessor.HttpContext?.User;
+            if (user == null || !await _authorizationService.AuthorizeAsync(user, Permissions.ManageMicrosoftAuthentication))
+            {
+                return null;
+            }
+
+            return Initialize<MicrosoftAccountSettingsViewModel>("MicrosoftAccountSettings_Edit", model =>
+            {
+                model.AppId = settings.AppId;
+                if (!string.IsNullOrWhiteSpace(settings.AppSecret))
```

Comment:
I prefer when the secret is always set, and the password masked. Otherwise as a user I never know if the password is already set, or it was cleared somehow. To make it possible, you'll need to set the `value` property in the input tag.

Reply:
if i understood correctly, i set empty AppSecret to model. I allready have the value attribute in input tag